bvm/ballerina-config/src/main/java/org/ballerinalang/config/ConfigRegistry.java

```diff
@@ -240,7 +240,7 @@ public long getAsInt(String key) {
                     return Long.parseLong((String) value);
                 }
                 return (Long) value;
-            } catch (ClassCastException e) {
+            } catch (ClassCastException | NumberFormatException e) {
                 throw new IllegalArgumentException(key + " does not map to a valid int");
```

Comment:
We are not interested in the exception(s) thrown here in this case because these will be due to user errors (e.g., setting a word as an env var and trying to read it as an int) and since those can be rectified by the user, we just simply throw an exception with a meaningful error message which will aid the user to correct his error.